triggers_windows.go

```diff
@@ -1,55 +1,204 @@
+			if err == ERROR_INVALID_PARAMETER {
+				grip.Debug(message.WrapError(err, message.Fields{
+					"id":      info.ID,
+					"pid":     info.PID,
+					"source":  mongodShutdownSignalTriggerSource,
+					"message": "did not open process because it has already exited",
+				}))
+			} else {
+				grip.Error(message.WrapError(err, message.Fields{
+					"id":      info.ID,
+					"pid":     info.PID,
+					"source":  mongodShutdownSignalTriggerSource,
+					"message": "failed to open process",
+				}))
+			}
 			return false
 		}
+		defer CloseHandle(proc)
 
-		pid := info.PID
-		eventName := mongodShutdownEventNamePrefix + strconv.Itoa(pid)
+		eventName := mongodShutdownEventNamePrefix + strconv.Itoa(info.PID)
 		utf16EventName, err := syscall.UTF16PtrFromString(eventName)
 		if err != nil {
-			grip.Errorf(errors.Wrapf(err, "failed to convert mongod shutdown event name '%s'", eventName).Error())
+			grip.Error(message.WrapError(err, message.Fields{
+				"id":      info.ID,
+				"pid":     info.PID,
+				"source":  mongodShutdownSignalTriggerSource,
+				"event":   eventName,
+				"message": "failed to convert event name",
+			}))
 			return false
 		}
 
 		event, err := OpenEvent(utf16EventName)
 		if err != nil {
-			grip.Errorf(errors.Wrapf(err, "failed to open event '%s'", eventName).Error())
+			if mongodExited(err) {
+				grip.Debug(message.WrapError(err, message.Fields{
+					"id":      info.ID,
+					"pid":     info.PID,
+					"source":  mongodShutdownSignalTriggerSource,
+					"event":   eventName,
+					"message": "did not open event because process has already exited",
+				}))
+			} else {
+				grip.Error(message.WrapError(err, message.Fields{
+					"id":      info.ID,
+					"pid":     info.PID,
+					"source":  mongodShutdownSignalTriggerSource,
+					"event":   eventName,
+					"message": "failed to open event",
+				}))
+			}
 			return false
 		}
 		defer CloseHandle(event)
 
 		if err := SetEvent(event); err != nil {
-			grip.Errorf(errors.Wrapf(err, "failed to signal event '%s'", eventName).Error())
+			if mongodExited(err) {
+				grip.Debug(message.WrapError(err, message.Fields{
+					"id":      info.ID,
+					"pid":     info.PID,
+					"source":  mongodShutdownSignalTriggerSource,
+					"event":   eventName,
+					"message": "did not signal event because process has already exited",
+				}))
+			} else {
+				grip.Error(message.WrapError(err, message.Fields{
+					"id":      info.ID,
+					"pid":     info.PID,
+					"source":  mongodShutdownSignalTriggerSource,
+					"event":   eventName,
+					"message": "failed to signal event",
+				}))
+			}
 			return false
 		}
 
+		waitStatus, err := WaitForSingleObject(proc, mongodShutdownEventTimeout)
+		if err != nil || waitStatus != WAIT_OBJECT_0 {
+			grip.Error(message.WrapError(err, message.Fields{
+				"id":      info.ID,
+				"message": "failed to wait on process",
+				"source":  mongodShutdownSignalTriggerSource,
+			}))
+			return false
+		}
+		if waitStatus != WAIT_OBJECT_0 {
```

Comment:
This branch will never execute, because the previous branch checks the same condition and returns.

Reply:
Removed the previous branch.